gapis/gfxapi/vulkan/dependency_graph.go

```diff
@@ -0,0 +1,1733 @@
+			readMemoryBindingsData(b, indirectBufMemoryBindings)
+			// Theoretically current attachments should be 'written' or 'modified' as
+			// the effect of this recorded command. However, after the Mutate() of
+			// VkQueueSubmit, the global state's LastBoundFramebuffer only records
+			// the last bound framebuffer and it can be different from the one used
+			// when this command is executed. So we rely on the last spot where
+			// attachments can be obtained through command parameters (not global
+			// state): vkCmdBeginRenderPass, to add the 'written' and 'modified'
+			// behavior to the attachments and the underlying data.
+			// Same logical for descriptors.
+		})
+
+	case *RecreateCmdDrawIndexedIndirect:
+		indirectBuf := typedAtom.Buffer
+		addRead(&b, g, vulkanStateKey(indirectBuf))
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+		recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+			// Theoretically vertex buffers and index buffer should be 'read' as the
+			// effect of a recorded command. Details in the handling of
+			// VkCmdBindVertexBuffers and VkCmdBindIndexBuffer
+			indirectBufMemoryBindings := getOverlappedBindingsForBuffer(indirectBuf)
+			readMemoryBindingsData(b, indirectBufMemoryBindings)
+			// Theoretically current attachments should be 'written' or 'modified' as
+			// the effect of this recorded command. However, after the Mutate() of
+			// VkQueueSubmit, the global state's LastBoundFramebuffer only records
+			// the last bound framebuffer and it can be different from the one used
+			// when this command is executed. So we rely on the last spot where
+			// attachments can be obtained through command parameters (not global
+			// state): vkCmdBeginRenderPass, to add the 'written' and 'modified'
+			// behavior to the attachments and the underlying data.
+			// Same logical for descriptors.
+		})
+
+	case *VkCmdDispatch:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+		// Theoretically writable descriptors should be 'read' and 'modified' as
+		// the effect of this recorded command. However, the
+		// CurrentComputePipeline in the global state after VkQueueSubmit's
+		// Mutate() is call, is only the last bound compute pipeline and can be
+		// different than the one used when this command is executed. So we rely
+		// on the last spot where descriptors can be obtained through command
+		// parameters (not global state): vkCmdBindDescriptorSets to add the
+		// 'modify' and 'read' behaviors.
+
+	case *RecreateCmdDispatch:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+		// Theoretically writable descriptors should be 'read' and 'modified' as
```

Comment:
Instead of copy-pasting this comment everywhere, we should maybe have it once at the top, but put it in the design document.